samples/interactive-tutorials/src/main/java/product/ImportProductsInlineSource.java

```diff
@@ -0,0 +1,218 @@
+            .setParent(DEFAULT_CATALOG)
+            .setInputConfig(inputConfig)
+            .build();
+
+    System.out.printf("Import products from inline source request: %s%n", importRequest);
+
+    return importRequest;
+  }
+
+  public static List<Product> getProducts() {
+    List<Product> products = new ArrayList<>();
+
+    Product product1;
+    Product product2;
+
+    float price1 = 16f;
+    float originalPrice1 = 45.0f;
+    float cost1 = 12.0f;
+
+    PriceInfo priceInfo1 =
+        PriceInfo.newBuilder()
+            .setPrice(price1)
+            .setOriginalPrice(originalPrice1)
+            .setCost(cost1)
+            .setCurrencyCode("USD")
+            .build();
+
+    ColorInfo colorInfo1 =
+        ColorInfo.newBuilder()
+            .addColorFamilies("Blue")
+            .addAllColors(Arrays.asList("Light blue", "Blue", "Dark blue"))
+            .build();
+
+    FulfillmentInfo fulfillmentInfo1 =
+        FulfillmentInfo.newBuilder()
+            .setType("pickup-in-store")
+            .addAllPlaceIds(Arrays.asList("store1", "store2"))
+            .build();
+
+    FieldMask fieldMask1 =
+        FieldMask.newBuilder()
+            .addAllPaths(Arrays.asList("title", "categories", "price_info", "color_info"))
+            .build();
+
+    // TO CHECK ERROR HANDLING COMMENT OUT THE PRODUCT TITLE HERE:
+    product1 =
+        Product.newBuilder()
+            .setTitle("#IamRemarkable Pen")
+            .setId(GENERATED_PRODUCT_ID)
+            .addAllCategories(Collections.singletonList("Office"))
+            .setUri(
+                "https://shop.googlemerchandisestore.com/Google+Redesign/"
+                    + "Office/IamRemarkable+Pen")
+            .addBrands("#IamRemarkable")
+            .setPriceInfo(priceInfo1)
+            .setColorInfo(colorInfo1)
+            .addFulfillmentInfo(fulfillmentInfo1)
+            .setRetrievableFields(fieldMask1)
+            .build();
+
+    float price2 = 35f;
+    float originalPrice2 = 45.0f;
+    float cost2 = 12.0f;
+
+    PriceInfo priceInfo2 =
+        PriceInfo.newBuilder()
+            .setPrice(price2)
+            .setOriginalPrice(originalPrice2)
+            .setCost(cost2)
+            .setCurrencyCode("USD")
+            .build();
+
+    ColorInfo colorInfo2 =
+        ColorInfo.newBuilder()
+            .addColorFamilies("Blue")
+            .addAllColors(Collections.singletonList("Sky blue"))
+            .build();
+
+    FulfillmentInfo fulfillmentInfo2 =
+        FulfillmentInfo.newBuilder()
+            .setType("pickup-in-store")
+            .addAllPlaceIds(Arrays.asList("store2", "store3"))
+            .build();
+
+    FieldMask fieldMask2 =
+        FieldMask.newBuilder()
+            .addAllPaths(Arrays.asList("title", "categories", "price_info", "color_info"))
+            .build();
+
+    product2 =
+        Product.newBuilder()
+            .setTitle("Android Embroidered Crewneck Sweater")
+            .setId(GENERATED_PRODUCT_ID)
```

Comment:
Should product 2 use a different ID than product 1?

Reply:
yes, you're right. fixed